staging/src/k8s.io/component-base/metrics/wrappers.go

```diff
@@ -65,6 +68,94 @@ type GaugeMetric interface {
 	SetToCurrentTime()
 }
 
+// PreContextGaugeMetric binds a Context to make a GaugeMetric
+type PreContextGaugeMetric interface {
+	// WithContext creates a GaugeMetric that uses the given Context
+	WithContext(ctx context.Context) GaugeMetric
+}
```

Comment:
This type isn't used anywhere else that the interface below. I'm somewhat against introducing interfaces for the sake of introducing them. Why can't we inline it below?